app/src/main/java/app/revanced/integrations/shared/Utils.java

```diff
@@ -722,6 +723,35 @@ public static void sortPreferenceGroups(@NonNull PreferenceGroup group) {
         }
     }
 
+    /**
+     * Set all preferences to multiline titles if the device is not using an English variant.
+     * The English strings are heavily scrutinized and all titles fit on screen
+     * except 2 or 3 preference strings and those do not affect readability.
+     *
+     * Allowing multiline for those 2 or 3 English preferences looks weird and out of place,
+     * and visually it looks better to clip the text and keep all titles 1 line.
+     */
+    @SuppressWarnings("deprecation")
+    public static void setPreferenceTitlesToMultiLineIfNeeded(PreferenceGroup group) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            return;
+        }
+
+        String deviceLanguage = Utils.getContext().getResources().getConfiguration().locale.getLanguage();
+        if (deviceLanguage.equals("en")) {
+            return;
+        }
+
+        for (int i = 0, prefCount = group.getPreferenceCount(); i < prefCount; i++) {
+            Preference pref = group.getPreference(i);
+            pref.setSingleLineTitle(false);
```

Comment:
Can that not be configured globally for all preferences?

Reply:
Seems like the  preference serializer could add `android:singleLineTitle="false"` according to https://stackoverflow.com/questions/19616238/how-to-wrap-a-preference-title-really.

Reply:
Shouldn't it be applied to any language? Currently, RYD and SB inject their own preference screens without the framework in the patches. So that should probably be unified.

Reply:
The wrapping should be agnostic to the strings. We may add new ones in the future or change strings so they get clipped. WIthout making them multiline we have to think about how to format them to avoid that and treat them exceptionally from other languages.